### PR TITLE
feat(docker): migrate to kork-credentials

### PR DESCRIPTION
**Pull request by nemesis-cruz-ar**
Implementing kork-credentials for docker registry accounts similar to k8s, cloudfoundry, aws, and ecs accounts.
This is the implementation of https://github.com/spinnaker/governance/blob/master/rfc/account-management.md for docker registry provider.

### CODE REVIEW

File: clouddriver-docker/src/main/java/com/netflix/spinnaker/config/DockerDefaultConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.netflix.spinnaker.credentials.CredentialsRepository;
+import com.netflix.spinnaker.credentials.MapBackedCredentialsRepository;
+import com.netflix.spinnaker.credentials.NoopCredentialsLifecycleHandler;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnProperty(
+    value = "dockerRegistry.enabled",
```
Comment (german-muzquiz) [inline clouddriver-docker/src/main/java/com/netflix/spinnaker/config/DockerDefaultConfiguration.java:31]: Spinnaker tries to enforce snake-case style for properties: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-docker/src/main/groovy/com/netflix/spinnaker/config/DockerRegistryConfiguration.groovy#L36

File: clouddriver-docker/src/test/java/com/netflix/spinnaker/clouddriver/docker/registry/security/DockerRegistryCredentialsLifecycleHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import com.netflix.spinnaker.clouddriver.docker.registry.DockerRegistryCloudProvider;
+import com.netflix.spinnaker.clouddriver.docker.registry.provider.DockerRegistryProvider;
+import com.netflix.spinnaker.clouddriver.docker.registry.provider.agent.DockerRegistryImageCachingAgent;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+@RunWith(JUnitPlatform.class)
+class DockerRegistryCredentialsLifecycleHandlerTest {
```
Comment (german-muzquiz) [inline clouddriver-docker/src/test/java/com/netflix/spinnaker/clouddriver/docker/registry/security/DockerRegistryCredentialsLifecycleHandlerTest.java:32]: Can you add these other tests:
* Credentials updated 
* One account credentials creates two caching agents

File: clouddriver-docker/clouddriver-docker.gradle [hunk 3]
```diff
@@ -9,6 +9,7 @@ dependencies {
 
   implementation "org.springframework.boot:spring-boot-actuator"
   implementation "org.springframework.boot:spring-boot-starter-web"
+  implementation "org.springframework.cloud:spring-cloud-context"
```
Comment (german-muzquiz) [inline clouddriver-docker/clouddriver-docker.gradle:12]: Is this dependency actually used for something?

**Review by german-muzquiz — COMMENTED** [review]
Thanks for adding kork credentials to docker provider! The changes look good, I only have minor comments.

I think the most important test is that for the same docker configuration, clouddriver behaves the same before and after this change, that is:

* Given two docker accounts with `cacheThreads: 2`:
  - 4 docker agents should be created that run periodically (you can verify that those docker agents are running because every 30 seconds or so they should print logs like `.d.r.p.a.DockerRegistryImageCachingAgent : Caching 7231 tagged images in {account}/DockerRegistryImageCachingAgent[1/2]`
  - 2 credentials of type `dockerRegistry` should be returned from `/credentials`

File: clouddriver-docker/clouddriver-docker.gradle (hunk 3 above)
Comment (nemesis-cruz-ar) [reply clouddriver-docker/clouddriver-docker.gradle:12]: we are using it for `@RefreshScope` annotation in
https://github.com/spinnaker/clouddriver/pull/5593/files#diff-392529703c92986db659767f07fbb4eca5e5d4c1f2603af40c5e0608d7f4bb91R56

File: clouddriver-docker/src/main/java/com/netflix/spinnaker/config/DockerDefaultConfiguration.java (hunk 1 above)
Comment (nemesis-cruz-ar) [reply clouddriver-docker/src/main/java/com/netflix/spinnaker/config/DockerDefaultConfiguration.java]: we changed the property name

File: clouddriver-docker/src/test/java/com/netflix/spinnaker/clouddriver/docker/registry/security/DockerRegistryCredentialsLifecycleHandlerTest.java (hunk 2 above)
Comment (nemesis-cruz-ar) [reply clouddriver-docker/src/test/java/com/netflix/spinnaker/clouddriver/docker/registry/security/DockerRegistryCredentialsLifecycleHandlerTest.java:33]: Done

**Comment by nemesis-cruz-ar** [comment]
demo for 2 accounts and 2 cache threads per account and credentials endpoint
multiple_accounts

demo for deploying to cloudfoundry using a docker registry account
deploy2CF

**Review by german-muzquiz — APPROVED** [review]
Nice demo videos!

**Pull request merged**